easybuild/easyconfigs/t/TRIQS/TRIQS-3.1.0-foss-2021b.eb

```diff
@@ -0,0 +1,70 @@
+easyblock = 'CMakeMake'
+
+name = 'TRIQS'
+version = '3.1.0'
+
+homepage = 'https://triqs.github.io/triqs'
+description = """
+ TRIQS (Toolbox for Research on Interacting Quantum Systems) is a
+ scientific project providing a set of C++ and Python libraries to
+ develop new tools for the study of interacting quantum systems.
+"""
+
+docurls = ['https://triqs.github.io/triqs/%(version_major_minor)s.x/']
+software_license = 'LicenseGPLv3'
+
+toolchain = {'name': 'foss', 'version': '2021b'}
+toolchainopts = {'pic': True, 'usempi': True}
+
+source_urls = ['https://github.com/TRIQS/triqs/releases/download/%(version)s/']
+sources = ['triqs-%(version)s.tar.gz']
+checksums = ['f1f358ec73498bc7ac3ed9665829d8605908f7f7fc876a5c2a01efe37d368f0e']
+
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+    ('Boost', '1.77.0'),
+    ('Clang', '13.0.1'),
+    ('Clang-Python-bindings', '13.0.1'),
+    ('GMP', '6.2.1'),
+    ('HDF5', '1.12.1'),
+    ('Mako', '1.1.4')
+]
+
+builddependencies = [
+    ('CMake', '3.22.1')
+]
+
+separate_build_dir = True
+
+# Remove installation directory before building. This fixes problems with
+# failing builds in the presence of preexisting installation.
+preconfigopts = "rm -rf %(installdir)s && "
+
+configopts = ""
```

Comment:
@Wentzell you removed the configopts here because `-DBuild_Deps=Always` is now the default.

looking at the dependencies that are built:
```
Cpp2Py
GTest
itertools
mpi
h5
nda
```
I'm wondering if this can be changed to `-DBuild_Deps=IfNotFound`?
HDF5 and MPI are already included as deps, and googletest (GTest) can be added easily too.

Reply:
Dear @smoors,

Yes we have switched to building the deps listed in [triqs/deps/CMakeLists.txt](https://github.com/TRIQS/triqs/blob/e24f9c8d38b4fea199a17cd240213803be81d629/deps/CMakeLists.txt#L63-L97) by default.
Note however that h5 and mpi refer to [TRIQS/h5](https://github.com/TRIQS/h5) and [TRIQS/mpi](https://github.com/TRIQS/mpi) which are only C++ Interfaces to the HDF5 and MPI C Libraries that are still searched for in the build environment. We don't won't to package those interfaces at the moment.

For GTest, which is only a build-time and not a runtime dependency, we would like to stick with the recommendation  to build it together with the project being tested.

Did the full build-test succeed? It seems the bot didn't report any test results?

Reply:
thanks for the reply.
can you remove the `configopts` as it's an empty string?

not sure why the bot did not report back. I'll rerun after you've made the change.

Reply:
Sure, I just changed this!